### PR TITLE
Fix for bug1018 and a refactored bug1012.

### PR DESCRIPTION
**Pull request by pmesnier**
The bug1018 regression test did not exactly replicate the error, but after extensive
research it seems the try rebind rule for corbaname isn't necessary after all.

### CODE REVIEW

**Comment by pmesnier** [comment]
turns out my previous comment is incorrect. A corbaname re-resolve is necessary after a connection loss or failure since a server restart may rebind in the name service, invalidating the previously resolved reference.

File: core/src/main/java/org/jacorb/orb/Delegate.java [hunk 1]
```diff
@@ -110,6 +110,7 @@
 public final class Delegate
     extends org.omg.CORBA_2_3.portable.Delegate
 {
+    private final String _identity;
```
Comment (rnc) [inline core/src/main/java/org/jacorb/orb/Delegate.java]: Is this needed?

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,5 @@ target
 .idea
 
 test/orbreinvoke/**/*~
+
+scratch
```
Comment (rnc) [inline .gitignore]: What are scratch files ? :-)

File: core/src/main/java/org/jacorb/orb/Delegate.java (hunk 1 above)
Comment (pmesnier) [inline core/src/main/java/org/jacorb/orb/Delegate.java]: probably not now.

File: .gitignore (hunk 2 above)
Comment (pmesnier) [inline .gitignore]: test run outputs, diffs, whatever. As I said in the squashed commit, transient gradeaux. I wanted to be sure I didn't accidentally git add that stuff.

**Pull request merged**